File: PlateformRH/src/main/java/com/PlateformRH/Employe/EmployeService.java
package com.PlateformRH.Employe;

import java.util.List;
import java.util.Map;

public interface EmployeService {

    employeDto updateEmploye(Long id, employeDto dto);

    employeDto getEmployeById(Long id);

    List<employeDto> getAllEmployes();

    void createEmploye(employe employe);

    void deleteEmploye(Long id);

    void prepareEmploye(employe employe);

    void delete(Long id);

    List<UtilisateurDTO> getAll();

    UtilisateurDTO update(Long id, UtilisateurDTO employe);

    List<UtilisateurDTO> getUsersByRole(String role);

    void changerMotDePasse(Long id, Map<String, String> body);

    void changerStatut(Long id);
}

File: rh/src/app/services/employe.service.ts
import { Injectable } from '@angular/core';

import { HttpClient } from '@angular/common/http';

import { Observable } from 'rxjs';

import { Employe,CreateEmploye } from '../models/employe';



@Injectable({
  providedIn: 'root'
})
export class EmployeService {

  private apiUrl = 'http://localhost:8000/employe';

  constructor(private http: HttpClient) { }

  // CREATE
  addEmploye(
    employe: CreateEmploye
  ): Observable<void> {

    return this.http.post<void>(
      this.apiUrl,
      employe
    );
  }

  // GET ALL
  getAllEmployes(): Observable<Employe[]> {

    return this.http.get<Employe[]>(
      this.apiUrl
    );
  }

  // GET BY ID
  getEmployeById(
    id: number
  ): Observable<Employe> {

    return this.http.get<Employe>(
      `${this.apiUrl}/${id}`
    );
  }

  // UPDATE
  updateEmploye(
    id: number,
    employe: Employe
  ): Observable<Employe> {

    return this.http.put<Employe>(
      `${this.apiUrl}/${id}`,
      employe
    );
  }

  // DELETE
  deleteEmploye(
    id: number
  ): Observable<string> {

    return this.http.delete(
      `${this.apiUrl}/${id}`,
      { responseType: 'text' }
    );
  }

  // GET SOLDES BY EMPLOYE
  getSoldesByEmploye(
    id: number
  ): Observable<any> {
    // For now, return mock data since backend might not have this endpoint
    return this.http.get(
      `${this.apiUrl}/${id}/soldes`
    );
  }

}
